dozer-ingestion/tests/test_suite/connectors/dozer.rs

```diff
@@ -0,0 +1,346 @@
+use std::collections::HashMap;
+
+use std::sync::Arc;
+use std::thread::JoinHandle;
+use std::time::Duration;
+
+use dozer_cli::shutdown::{self, ShutdownSender};
+use dozer_cli::simple::SimpleOrchestrator;
+use dozer_ingestion::connectors::dozer::NestedDozerConnector;
+use dozer_ingestion::connectors::{CdcType, SourceSchema};
+use dozer_types::grpc_types::ingest::ingest_service_client::IngestServiceClient;
+use dozer_types::grpc_types::ingest::{IngestRequest, OperationType};
+use dozer_types::grpc_types::types::{value, DurationType, PointType, Record, Value};
+use dozer_types::ingestion_types::GrpcConfig;
+use dozer_types::log::info;
+use dozer_types::models::api_endpoint::ApiEndpoint;
+use dozer_types::models::source::Source;
+use dozer_types::types::{Field, FieldDefinition, FieldType};
+use dozer_types::{
+    ingestion_types::{NestedDozerConfig, NestedDozerLogOptions},
+    models::api_config::AppGrpcOptions,
+};
+
+use futures::lock::Mutex;
+use tempdir::TempDir;
+use tokio::runtime::Runtime;
+use tonic::async_trait;
+use tonic::transport::Channel;
+
+use crate::test_suite::records::Operation;
+use crate::test_suite::{
+    CudConnectorTest, DataReadyConnectorTest, FieldsAndPk, InsertOnlyConnectorTest,
+};
+
+pub struct DozerConnectorTest {
+    _tmpdir: TempDir,
+    shutdown: Option<(JoinHandle<()>, ShutdownSender)>,
+    ingest_state: Arc<Mutex<(u32, IngestServiceClient<Channel>)>>,
+    schema: String,
+}
+
+fn field_to_grpc(field: Field) -> Option<value::Value> {
```

Comment:
I think we have this function somewhere. May just have to make that function public.